plugins/modules/idrac_license.py

```diff
@@ -0,0 +1,577 @@
+    else:
+        license_file_name = f"{module.params['license_id']}_iDRAC_license.txt"
+    license_status = self.idrac.invoke_request(export_license_url, "POST", data=payload)
+    license_data = license_status.json_data
+    license_file = license_data.get("LicenseFile")
+    file_name = os.path.join(path, license_file_name)
+    with open(file_name, "w") as fp:
+        fp.writelines(license_file)
+    return license_status
+
+
+def export_license_http(self, module):
+    uri = validate_and_get_first_resource_id_uri(
+        self.module, self.idrac, SYSTEMS_URI)
+    export_license_url = f"{uri[0]}/{EXPORT_LICENCE_NETWORK_SHARE_URI}"
+    payload = {}
+    payload["EntitlementID"] = module.params.get('license_id')
+    if module.params.get('share_parameters').get('share_type') == "http":
+        payload["ShareType"] = "HTTP"
+    else:
+        payload["ShareType"] = "HTTPS"
+    payload["IPAddress"] = module.params.get('share_parameters').get('ip_address')
+    payload["ShareName"] = module.params.get('share_parameters').get('share_name')
+    payload["UserName"] = module.params.get('share_parameters').get('username')
+    payload["Password"] = module.params.get('share_parameters').get('password')
+    payload["IgnoreCertWarning"] = IGNORE_CERTIFICATE_WARNING[module.params.get('share_parameters').get('ignore_certificate_warning')]
+    if module.params.get('share_parameters').get('proxy_support') == "parameters_proxy":
+        payload["ProxySupport"] = PROXY_SUPPORT[module.params.get('share_parameters').get('proxy_support')]
+        payload["ProxyType"] = PROXY_TYPE[module.params.get('share_parameters').get('proxy_type')]
+        payload["ProxyServer"] = module.params.get('share_parameters').get('proxy_server')
+        payload["ProxyPort"] = module.params.get('share_parameters').get('proxy_port')
+        if module.params.get('share_parameters').get('proxy_username') and module.params.get('share_parameters').get('proxy_password'):
+            payload["ProxyUname"] = module.params.get('share_parameters').get('proxy_username')
+            payload["ProxyPasswd"] = module.params.get('share_parameters').get('proxy_password')
+    export_status = export_license(self, module, payload, export_license_url)
+    return export_status
+
+
+def export_license_cifs(self, module):
+    uri = validate_and_get_first_resource_id_uri(
+        self.module, self.idrac, SYSTEMS_URI)
+    export_license_url = f"{uri[0]}/{EXPORT_LICENCE_NETWORK_SHARE_URI}"
+    payload = {}
+    payload["EntitlementID"] = module.params.get('license_id')
+    payload["ShareType"] = "CIFS"
+    if module.params.get('share_parameters').get('workgroup'):
+        payload["Workgroup"] = module.params.get('share_parameters').get('workgroup')
+    share_details = get_share_details(module)
+    payload.update(share_details)
+    export_status = export_license(self, module, payload, export_license_url)
+    return export_status
+
+
+def export_license_nfs(self, module):
```

Comment:
move this inside export license class. any reason why kept outside of class?

Reply:
Done